sql/expression/function/find_in_set.go

```diff
@@ -0,0 +1,119 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+// FindInSet takes out the specified unit(s) from the time expression.
+type FindInSet struct {
+	expression.BinaryExpression
+}
+
+var _ sql.FunctionExpression = (*FindInSet)(nil)
+var _ sql.CollationCoercible = (*FindInSet)(nil)
+
+// NewFindInSet creates a new FindInSet expression.
+func NewFindInSet(e1, e2 sql.Expression) sql.Expression {
+	return &FindInSet{
+		expression.BinaryExpression{
+			Left:  e1,
+			Right: e2,
+		},
+	}
+}
+
+// FunctionName implements sql.FunctionExpression
+func (f *FindInSet) FunctionName() string {
+	return "find_in_set"
+}
+
+// Description implements sql.FunctionExpression
+func (f *FindInSet) Description() string {
+	return "returns a value in the range of 1 to N if the string str is in the string list strlist consisting of N substrings"
+}
+
+// Type implements the Expression interface.
+func (f *FindInSet) Type() sql.Type { return types.Int64 }
+
+// CollationCoercibility implements the interface sql.CollationCoercible.
+func (*FindInSet) CollationCoercibility(ctx *sql.Context) (collation sql.CollationID, coercibility byte) {
+	return sql.Collation_binary, 5
+}
+
+func (f *FindInSet) String() string {
+	return fmt.Sprintf("%s(%s from %s)", f.FunctionName(), f.Left, f.Right)
+}
+
+// WithChildren implements the Expression interface.
+func (f *FindInSet) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != 2 {
+		return nil, sql.ErrInvalidChildrenNumber.New(f, len(children), 2)
+	}
+	return NewFindInSet(children[0], children[1]), nil
+}
+
+// Eval implements the Expression interface.
+func (f *FindInSet) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	if f.Left == nil || f.Right == nil {
+		return nil, nil
+	}
+
+	left, err := f.Left.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	right, err := f.Right.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	if left == nil || right == nil {
+		return nil, nil
+	}
+
+	lVal, _, err := types.LongText.Convert(left)
+	if err != nil {
+		return nil, err
+	}
+	l := strings.ToLower(lVal.(string))
+
+	// always returns 0 when left contains a comma
+	if strings.Contains(l, ",") {
+		return 0, nil
+	}
+
+	// TODO: SET type uses binary arithmetic?
```

Comment:
It seems like you're pretty close to supporting set types. If `right`'s type is a SetType, then you could grab right's value and call `right.Type().BitsToString(rVal);` or something, to get a comma separated string representation of the bitmap values mapped back to the set strings. 

This TODO currently isn't going to be very helpful to future code readers though. Try seeing if you can get the simple approach for set types to work here and then this TODO could be more helpful if it were something like "Optimize set type lookup to work off of bitmap directly" or something. That would give future readers a better clue about what needs to be done in the future.